agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -2406,25 +2379,47 @@ bool slave_thread::handle_cmdu_ap_manager_message(
             notification_out->mac() = notification_in->params().mac;
             message_com::send_cmdu(monitor_socket, cmdu_tx);
         }
+
         //notify master
-        // if the master is not connected, remove element from pending_client_association
         if (master_socket) {
-            auto notification_out = message_com::create_vs_message<
-                beerocks_message::cACTION_CONTROL_CLIENT_DISCONNECTED_NOTIFICATION>(
-                cmdu_tx, beerocks_header->id());
+            LOG(DEBUG) << "Controller is not connected";
+            return true;
+        }
 
-            if (notification_out == nullptr) {
-                LOG(ERROR)
-                    << "Failed building ACTION_CONTROL_CLIENT_DISCONNECTED_NOTIFICATION message!";
-                break;
-            }
-            notification_out->params() = notification_in->params();
-            send_cmdu_to_controller(cmdu_tx);
+        // build 1905.1 message CMDU to send to the controller
+        if (!cmdu_tx.create(0, ieee1905_1::eMessageType::TOPOLOGY_NOTIFICATION_MESSAGE)) {
+            LOG(ERROR) << "cmdu creation of type TOPOLOGY_NOTIFICATION_MESSAGE, has failed";
+            return false;
+        }
 
-        } else {
-            pending_client_association_cmdu.erase(client_mac);
+        auto client_association_event_tlv = cmdu_tx.addClass<wfa_map::tlvClientAssociationEvent>();
+        if (!client_association_event_tlv) {
+            LOG(ERROR) << "addClass tlvClientAssociationEvent failed";
+            return false;
+        }
+        client_association_event_tlv->client_mac() = notification_in->params().mac;
+        client_association_event_tlv->bssid()      = notification_in->params().bssid;
+        client_association_event_tlv->association_event() =
+            wfa_map::tlvClientAssociationEvent::CLIENT_HAS_LEFT_THE_BSS;
+
```

Comment:
I would feel a little bit more comfortable if the unused fields would be set explicitly, either here or by specifying defaults in the yaml file.

Reply:
The capability struct has default values, and on every CMDU creation the whole buffer is set to zero, so this is not needed.